### PR TITLE
feat: add table id and engine to information_schema.TABLES

### LINKED ISSUE #1355: Add table id and engine to information_schema.tables
### What type of enhancement is this?

API improvement, User experience

### What does the enhancement do?

#1327 introduces the `tables` table to the information schema. Currently, this table contains the following columns
- table_catalog
- table_schema
- table_name
- table_type

It's helpful to provide more information about the table, including
- the table id (TABLE_ID or GREPTIME_TABLE_ID)
- the engine name (ENGINE)

MySQL has an `ENGINE` column in its [`tables`](https://dev.mysql.com/doc/refman/8.0/en/information-schema-tables-table.html) table. TiDB adds a `TIDB_TABLE_ID` [to show the table's id](https://docs.pingcap.com/tidb/dev/information-schema-tables).

### Implementation challenges

We need to add more columns to the `tables` table
https://github.com/GreptimeTeam/greptimedb/blob/09f003d01d8e1c7cf54b3a458675dc5cda41780a/src/catalog/src/information_schema/tables.rs#L41-L47

Then refactor the `add_table()` method to put more information to the output record batch.
https://github.com/GreptimeTeam/greptimedb/blob/09f003d01d8e1c7cf54b3a458675dc5cda41780a/src/catalog/src/information_schema/tables.rs#L101-L104

Table id and the engine are available in the `TableInfo`.

### PR DESCRIPTION
**Pull request by haohuaijin**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

add `table_id` and `table_engine` to information_schema.TABLES

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
close #1355

### CODE REVIEW

File: src/catalog/src/information_schema/tables.rs [hunk 1]
```diff
@@ -44,6 +44,8 @@ impl InformationSchemaTables {
             ColumnSchema::new("table_schema", ConcreteDataType::string_datatype(), false),
             ColumnSchema::new("table_name", ConcreteDataType::string_datatype(), false),
             ColumnSchema::new("table_type", ConcreteDataType::string_datatype(), false),
+            ColumnSchema::new("table_id", ConcreteDataType::uint32_datatype(), true),
+            ColumnSchema::new("table_engine", ConcreteDataType::string_datatype(), true),
```
Comment (evenyag) [inline src/catalog/src/information_schema/tables.rs:48]: MySQL has an `ENGINE` column in this table, I think we can reuse that column.
```suggestion
            ColumnSchema::new("engine", ConcreteDataType::string_datatype(), true),
```

**Review by waynexia — COMMENTED** [review]
Mostly looks good to me 👍  We can add some tests about information schema to sqlness to verify and prevent future breakings.

File: tests/cases/standalone/common/system/information_schema.result [hunk 2]
```diff
@@ -27,18 +27,18 @@ order by table_name;
...
-| table_catalog | table_schema       | table_name | table_type |
-+---------------+--------------------+------------+------------+
-| greptime      | information_schema | tables     | VIEW       |
-| greptime      | my_db              | foo        | BASE TABLE |
-+---------------+--------------------+------------+------------+
++---------------+--------------------+------------+------------+----------+--------+
+| table_catalog | table_schema       | table_name | table_type | table_id | engine |
++---------------+--------------------+------------+------------+----------+--------+
+| greptime      | information_schema | tables     | VIEW       |          |        |
+| greptime      | my_db              | foo        | BASE TABLE | 1024     | mito   |
```
[the remark this answers is not in the document]
Comment (WenyXu) [reply tests/cases/standalone/common/system/information_schema.result:40]: How about putting this test into `frontend::tests::instance_test.rs` 🤔

[the remark this answers is not in the document]
Comment (haohuaijin) [reply tests/cases/standalone/common/system/information_schema.result]: Good idea. But the `frontend::tests::instance_test.rs` also have this test, I just update it.

File: src/frontend/src/tests/instance_test.rs [hunk 3]
```diff
@@ -913,27 +913,27 @@ async fn test_information_schema(instance: Arc<dyn MockInstance>) {
...
-+---------------+--------------------+------------+------------+
-| greptime      | public             | numbers    | BASE TABLE |
-| greptime      | public             | scripts    | BASE TABLE |
-| greptime      | information_schema | tables     | VIEW       |
-+---------------+--------------------+------------+------------+";
++---------------+--------------------+------------+------------+----------+-------------+
+| table_catalog | table_schema       | table_name | table_type | table_id | engine      |
++---------------+--------------------+------------+------------+----------+-------------+
+| greptime      | public             | numbers    | BASE TABLE | 1        | test_engine |
+| greptime      | public             | scripts    | BASE TABLE | 1        | mito        |
```
Comment (haohuaijin) [inline src/frontend/src/tests/instance_test.rs:924]: In local machine the scripts's `table_id` is 1, but in github test the `table_id` is 1024.

Comment (WenyXu) [reply src/frontend/src/tests/instance_test.rs:924]: I checked the CI log. Did you check it passed the `instance_test::test_information_schema::case_2_test_with_distributed` test? (If you are using the vs code, you will find multiple `run test` text buttons) . 

```
  PASS [   0.074s] frontend tests::instance_test::test_information_schema::case_1_test_with_standalone
   FAIL [   1.437s] frontend tests::instance_test::test_information_schema::case_2_test_with_distributed
```

I guess it's due to the inconsistent catalog behaviors between the distributed and standalone modes.

Comment (evenyag) [reply src/frontend/src/tests/instance_test.rs:924]: Looks like this test runs the same cases both for standalone and distributed mode. What about ignoring the table id column in the tests? The actual value of the table id in the test is not so significant now.

Comment (MichaelScofield) [reply src/frontend/src/tests/instance_test.rs]: You can add a method to `MockInstance` to return whether it is distributed like this: https://github.com/MichaelScofield/greptimedb/blob/1380772cc38c6dea4b6e5ae132680fecc7b97b41/src/frontend/src/tests/test_util.rs#L30

And set different expected values for each mode: https://github.com/MichaelScofield/greptimedb/blob/1380772cc38c6dea4b6e5ae132680fecc7b97b41/src/frontend/src/tests/instance_test.rs#L328

**Review by MichaelScofield — APPROVED** [review]

**Review by evenyag — APPROVED** [review]

**Pull request merged**